receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper.go

```diff
@@ -31,8 +31,8 @@ const metricsLen = 1
 
 // scraper for CPU Metrics
 type scraper struct {
-	config    *Config
-	startTime pdata.Timestamp
+	config *Config
+	mb     *metadata.SystemCPUTimeMetricBuilder
```

Comment:
I think this is looking great, and agree with w/ Bogdan's suggested roadmap. 

In addition, I hope that we can eventually have one builder per scraper instance. Perhaps then this builder can be partially managed by the scraper controller.

```go
type scraper struct {
    ...
    mb *metadata.MetricBuilder
}

func (s *scraper) start(context.Context, component.Host) error {
    // maybe scraper controller initializes this
    s.mb = metadata.NewMetricBuilder(withStartTime)
}

func (s *scraper) scrape(_ context.Context) (pdata.Metrics, error) {
    cpuTime := gopsutil....

    now := time.Now()
    s.mb.CollectCPUTimeDataPoint(now, cpuTime.CPU, metadata.LabelState.User, cpuTime.User)
    s.mb.CollectOtherMetricDataPoint(now, cpuTime.CPU, metadata.LabelState.User, cpuTime.User)

    s.mb.Emit() // maybe called automatically by scraper controller
}
```

This isn't all that different than what we currently have in this PR, but as a scraper author I think it would be more intuitive because I am less focused on managing builders.

Reply:
I like this suggestion. We probably can start with this approach and see if there is a need for per-metric builders going forward.

@bogdandrutu WDYT?

Reply:
I like the one builder/helper for everything. Some suggestions:
* Collect -> Record
* Emit must be called by the scraper, because the scraper framework must work for scrapers without metadata (scrapers that don't know in advance the metrics they will collect).